tests/functional/test_environment.py

```diff
@@ -10,13 +10,17 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import os
 import platformdirs
 
 
 def test_xdg_environment():
     # Backstop checks for Warehouse's (Dockerfile-configured) environment.
-    user_data = platformdirs.user_data_dir()
-    user_cache = platformdirs.user_cache_dir()
+    # NOTE: We don't check the exact paths here, because they vary between
+    # environments: they're in `$HOME` in local and CI builds, and in `/tmp`
+    # when deployed.
+    user_data = platformdirs.user_data_dir(ensure_exists=True)
+    user_cache = platformdirs.user_cache_dir(ensure_exists=True)
```

Comment:
NB: With `ensure_exists` this is now partially a no-op test, at least in local/CI testing (since these dirs don't exist). However for Cabotage/deployment it'll end up testing something useful, per https://github.com/cabotage/cabotage-app/pull/85

Reply:
just for clarity, we do not execute tests in the cabotage environment. should this be a runtime check?

Reply:
Ah, didn't realize that -- yeah, in that case it probably makes more sense as a runtime check. I'll look at putting it somewhere early in `python -m warehouse`'s initialization 🙂

Reply:
I've put this in `warehouse/config.py`, as part of the early initialization that happens there. But I'm not 100% sure if that's the best place 🙂

Reply:
In smoke testing this, it looks like this doesn't correctly detect that things are working.

I rolled out a new release of test.pypi.org using this wrapper and you can see results in https://github.com/cabotage/cabotage-app/pull/85#issuecomment-2243361188

`platformdirs.user_data_dir(ensure_exists=True)` succeeds, but `platformdirs.user_data_dir('app', ensure_exists=True)` fails, indicating that this wouldn't actually ensure success for xdg/platformdirs use later on.

Reply:
appears this is due to the existing /tmp/{share, cache} files created by the Dockerfile on main :)

Reply:
validated this locally!

I still think it may be wise to change these to `platformdirs.user_cache_dir(secrets.token_urlsafe(), ensure_exists=True)` and similar to ensure not only that the dirs exist but that they're writable.

Reply:
> appears this is due to the existing /tmp/{share, cache} files created by the Dockerfile on main :)

Whoops 🤦 -- I _think_ merging this first should fix that then, since we won't be doing `mkdir` in the Dockerfile anymore.

Reply:
> I still think it may be wise to change these to `platformdirs.user_cache_dir(secrets.token_urlsafe(), ensure_exists=True)` and similar to ensure not only that the dirs exist but that they're writable.

Done!